src/TritonRoute/src/TritonRoute.cpp

```diff
@@ -326,6 +332,51 @@ void TritonRoute::readParams(const string& fileName)
   }
 }
 
+void TritonRoute::setParams(const string& guide_file,
+                            const string& output_guide_file,
+                            const string& output_maze_file,
+                            const string& output_DRC_file,
+                            const string& output_CMap_file,
+                            const string& dbProcessNode,
+                            int drouteEndIterNum,
+                            int drouteViaInPinBottomLayerNum,
+                            int drouteViaInPinTopLayerNum,
+                            int or_seed,
+                            double or_k,
+                            int bottomRoutingLayer,
+                            int topRoutingLayer,
+                            int initRouteShapeCost,
+                            int verbose)
+{
+  GUIDE_FILE = guide_file;
+  OUTGUIDE_FILE = output_guide_file;
+  OUT_MAZE_FILE = output_maze_file;
+  DRC_RPT_FILE = output_DRC_file;
+  CMAP_FILE = output_CMap_file;
+  VERBOSE = verbose;
+  DBPROCESSNODE = dbProcessNode;
+  if (drouteViaInPinBottomLayerNum > 0) {
+    VIAINPIN_BOTTOMLAYERNUM = drouteViaInPinBottomLayerNum;
+  }
+  if (drouteViaInPinTopLayerNum > 0) {
+    VIAINPIN_TOPLAYERNUM = drouteViaInPinTopLayerNum;
+  }
+  if (drouteEndIterNum > 0) {
+    END_ITERATION = drouteEndIterNum;
+  }
+  OR_SEED = or_seed;
+  OR_K = or_k;
+  if (bottomRoutingLayer > 0) {
+    BOTTOM_ROUTING_LAYER_NAME = bottomRoutingLayer;
+  }
+  if (topRoutingLayer > 0) {
+    TOP_ROUTING_LAYER_NAME = topRoutingLayer;
+  }
+  if (initRouteShapeCost > 0) {
+    ROUTESHAPECOST = static_cast<frUInt4>(initRouteShapeCost);
```

Comment:
Do we have a tcl function to check the bounds of an integer so that it doesn't get truncated?

Reply:
Not that I know of but I can't imaging anyone setting the cost high enough for it to truncate.